src/Microsoft.AspNetCore.Razor.Language/TagHelperDescriptorComparer.cs

```diff
@@ -62,6 +62,7 @@ public virtual bool Equals(TagHelperDescriptor descriptorX, TagHelperDescriptor
             return descriptorX != null &&
                 string.Equals(descriptorX.Kind, descriptorY.Kind, StringComparison.Ordinal) &&
                 string.Equals(descriptorX.AssemblyName, descriptorY.AssemblyName, StringComparison.Ordinal) &&
+                string.Equals(descriptorX.Name, descriptorY.Name, StringComparison.Ordinal) &&
```

Comment:
This was a bug BTW, the name wasn't compared here.

Reply:
Unrelated - for all of the collection comparisons below, we could compare their sizes first since they're all (?) IReadOnlyList types.

Reply:
I think we should wait for data - or add a perf test.